docs/user/guides/setup/configuration-file.rst

```diff
@@ -0,0 +1,141 @@
+How to add a configuration file
+===============================
+
+As part of the initial setup for your Read the Docs site,
+you need to create a **configuration file** called ``.readthedocs.yaml``.
+The configuration file tells Read the Docs what specific settings to use for your project.
+
+This how-to guide covers:
+
+#. Where to put your configuration file.
+#. Templates to setup your configuration file from scratch.
+#. The most basic sections in the configuration file that can help you get started.
+#. Next steps
+
+.. seealso::
+
+   :doc:`/tutorial/index`.
+     Following the steps in our tutorial will help you setup your first documentation project.
+
+
+Where to put your configuration file
+------------------------------------
+
+The ``.readthedocs.yaml`` file should be placed in the top-most directory of your project's repository.
+We will get to the contents of the file in the next steps.
+
+When you have changed the configuration file,
+you need to commit and push the changes to your Git repository.
+Read the Docs will then automatical find and use the configuration to build your project.
+
+Getting started with a template
+-------------------------------
+
+Here are some variations of the configuration file that can help you get started.
+Pick the one that resembles your project the most,
+copy its contents to ``.readthedocs.yaml`` and add the file to your Git repository.
+
+.. tabs::
+
+    .. tab:: Sphinx
+
+        If your project uses Sphinx,
+        we offer a special builder optimized for Sphinx projects.
+
+        .. literalinclude:: /config-file/examples/sphinx/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+
+    .. tab:: MkDocs
+
+        If your project uses MkDocs,
+        we offer a special builder optimized for MkDocs projects.
+
+        .. literalinclude:: /config-file/examples/mkdocs/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+    .. tab:: Other tools
+
+        If you are using another documentation tool,
+        you can configure the commands of your documentation tool in the following code.
+
+        You will need to create the output directories and direct your documentation tool to write its outputs into those directories.
+
+        .. literalinclude:: /config-file/examples/custom-commands-echo/.readthedocs.yaml
+           :language: yaml
+           :linenos:
+
+
+Editing the template
+--------------------
```

Comment:
We should decide how this section should work.

Two alternatives:

1. Provide a list of sections that are of special interest for the new user (how it is now)
2. Simply provide guidance about the fact that the file has to be edited and then reference _the reference_.

Reply:
If we go with option 1), we should add some more sections. Currently, we are just explaining two parts :) We should definitely also explain the `build.os` section.

Reply:
Well, what I think we need to make our reference more usable is some better content _summarizing_ the configuration file.

This page will serve as a good place for the user to jump off into deeper content, and so I might still lean towards being more thorough with the number of configuration options we mention here, but not going into as much detail as we have.

For example, we describe the python key and what it is, and the python.install.requirements key and what it is below. Instead, I think I'd find the most use out of this page if it summarized the Python install options, but pointed to the relevant config reference for all details.

So, instead of the level of detail we have now, something closer to:

```rst
There are several configuration file options for installing your Python dependencies:

- Use :ref:`python.install.package` to install a package using pip
- Use :ref:`python.install.requirements` to install packages from a requirements file
- Use :ref:`build.jobs` to install packages using Poetry or PDM
```

Reply:
Yup, I totally agree with that.

At the point of reading that section, the user should be familiar with changing content in the config file, so it should just be a question of introducing some important reference items.

Reply:
Thanks for the copy-paste friendly suggestion :heart:

Reply:
Oops, well I was writing in pseudo documentation, so I don't know how well it will paste :upside_down_face:

Reply:
Yes but it's really nice with a clear example, copy-pasting and then adapting the example will be easier :)

Reply:
Added a structure now. I found it hard to simply condense all possible adjustments in a simple list and opted for headings and `seealso::` boxes.